tests/5.1/target/test_target_memcpy_async_depobj.c

```diff
@@ -0,0 +1,76 @@
+//===--- test_target_memcpy_async_depobj.c ----------------------------===//
+//
+//  Inspired from OpenMP 5.1 Examples Doc, 5.16.4 & 8.9
+//  This test utilizes the omp_target_memcpy_async construct to
+//  allocate memory on the device asynchronously. The construct
+//  uses 'obj' for dependency, so that memory is only copied once
+//  the variable listed in the depend clause is changed.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+int errors, i;
+
+int test_target_memcpy_async_depobj() {
+
+    int h, t, i;
+    errors = 0;
+    double *mem;
+    double *mem_dev_cpy;
+    h = omp_get_initial_device();
+    t = omp_get_default_device();
+
+
+    mem = (double *)malloc( sizeof(double)*N);
+    mem_dev_cpy = (double *)omp_target_alloc( sizeof(double)*N, t);
+
+    OMPVV_TEST_AND_SET_VERBOSE(errors, mem_dev_cpy == NULL);
+
+    for(i = 0; i < N; i++){
+        mem[i] = i;
+    }
+    omp_depend_t obj;
+    #pragma omp depobj(obj) depend(inout: mem_dev_cpy)
+    omp_depend_t obj_arr[1] = {obj};
+
+    /* copy to device memory */
+    omp_target_memcpy_async(mem_dev_cpy, mem, sizeof(double)*N,
+                                0,          0,
+                                t,          h,
+                                1,          obj_arr[0]);
+
+    #pragma omp taskwait depend(depobj: obj)
+    #pragma omp target is_device_ptr(mem_dev_cpy) device(t) depend(depobj: obj)
+    for(i = 0; i < N; i++){
+        mem_dev_cpy[i] = mem_dev_cpy[i]*2; // initialize data
+    }
+
+    /* copy to host memory */
+    omp_target_memcpy_async(mem, mem_dev_cpy, sizeof(double)*N,
+                                0,          0,
+                                h,          t,
+                                1,          obj_arr[0]);
```

Comment:
Fails here - and below with the message:
```
test_target_memcpy_async_depobj.c: In function ‘test_target_memcpy_async_depobj’:
test_target_memcpy_async_depobj.c:47:52: error: incompatible type for argument 9 of ‘omp_target_memcpy_async’
   47 |                                 1,          obj_arr[0]);
      |                                             ~~~~~~~^~~
      |                                                    |
      |                                                    omp_depend_t
In file included from test_target_memcpy_async_depobj.c:11:
....lib/gcc/x86_64-pc-linux-gnu/13.0.0/include/omp.h:277:42: note: expected ‘omp_depend_t *’ but argument is of type ‘omp_depend_t’
  277 |                                     int, omp_depend_t *)
```
Either of the following should work and should be identical under the hood:
```suggestion
                                1,          &obj_arr[0]);
```
or
```suggestion
                                1,          obj_arr);
```
_Side remark: Since today, GCC mainline (alias GCC 13, to be released in a year) supports `omp_target_memcpy_async` and `omp_target_memcpy_rect_async`_